src/test/java/com/mongodb/hibernate/jdbc/MongoConnectionProviderTests.java

```diff
@@ -177,18 +187,54 @@ private void verifyMongoClient(
         }
     }
 
-    private SessionFactory buildSessionFactory(MongoClientCustomizer mongoClientCustomizer, String connectionString) {
+    private SessionFactory buildSessionFactory(
+            MongoDialectConfigurator mongoDialectConfigurator, String connectionString) {
         var standardServiceRegistryBuilder = new StandardServiceRegistryBuilder();
-        if (mongoClientCustomizer != null) {
-            standardServiceRegistryBuilder.addService(MongoClientCustomizer.class, mongoClientCustomizer);
+        if (mongoDialectConfigurator != null) {
+            standardServiceRegistryBuilder.addService(MongoDialectConfigurator.class, mongoDialectConfigurator);
         }
         if (connectionString != null) {
-            standardServiceRegistryBuilder.applySetting(JdbcSettings.JAKARTA_JDBC_URL, connectionString);
+            standardServiceRegistryBuilder.applySetting(JAKARTA_JDBC_URL, connectionString);
         }
         return new MetadataSources(standardServiceRegistryBuilder.build())
                 .getMetadataBuilder()
                 .build()
                 .getSessionFactoryBuilder()
                 .build();
     }
+
+    private static final class NoopServiceRegistryImplementor implements ServiceRegistryImplementor {
```

Comment:
Mockito fails to mock `ServiceRegistryImplementor` despite the latter being an interface (it mocks other interfaces just fine). I failed to quickly figure out what to do with that, and decided that implementing a noop implementation manually is an OK compromiss.

Reply:
This is interesting. The first time I ever encountered such Mockito issue.

A better alternative to assert that `autoCommit` is true by default is to verify in integration testing, which is more compelling and intuitive. E.g. we could add the following method to `SessionTests` in integrationTest folder:

```
    @Test
    void defaultAutoCommitTrue() {
        assertDoesNotThrow(() -> session.doWork(connection -> {
            assertTrue(connection.getAutoCommit());
        }));
    }
```

Reply:
Thank you for helping me with a solution.

Reply:
> A better alternative to assert that autoCommit is true by default is to verify in integration testing

A test via `Session` does not test what I am testing here. Only be testing the provider directly can we verify that the connections it produces have auto-commit enabled.